packages/aws-cdk/lib/version.ts

```diff
@@ -0,0 +1,98 @@
+import { exec as _exec } from 'child_process';
+import colors = require('colors/safe');
+import { open as _open, stat as _stat } from 'fs';
+import semver = require('semver');
+import { promisify } from 'util';
+import { debug, print, warning } from '../lib/logging';
+
+const ONE_DAY_IN_SECONDS = 1 * 24 * 60 * 60;
+
+const exec = promisify(_exec);
+const open = promisify(_open);
+const stat = promisify(_stat);
+
+export const DISPLAY_VERSION = `${versionNumber()} (build ${commit()})`;
+
+function versionNumber(): string {
+  return require('../package.json').version.replace(/\+[0-9a-f]+$/, '');
+}
+
+function commit(): string {
+  return require('../build-info.json').commit;
+}
+
+export class CacheFile {
+  private readonly file: string;
+
+  // File modify times are accurate only till the second, hence using seconds as precision
+  private readonly ttlSecs: number;
+
+  constructor(file: string, ttlSecs: number) {
+    this.file = file;
+    this.ttlSecs = ttlSecs;
+  }
+
+  public async hasExpired(): Promise<boolean> {
+    try {
+      const lastCheckTime = (await stat(this.file)).mtimeMs;
+      const today = new Date().getTime();
+
+      if ((today - lastCheckTime) / 1000 > this.ttlSecs) { // convert ms to secs
+        return true;
+      }
+      return false;
+    } catch (err) {
+      if (err.code === 'ENOENT') {
+        return true;
+      } else {
+        throw err;
+      }
+    }
+  }
+
+  public async update(): Promise<void> {
+    await open(this.file, 'w');
```

Comment:
This leaks an FD. Please close it explicitly if you intend this to just be a `touch` operation.

Reply:
Fixed now.